File: problems_list/0429.N叉树的层序遍历/0429-N叉树的层序遍历.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        if(root==nullptr){
            return {};
        }

        vector<vector<int>> ans;
        queue<Node*> q;
        q.push(root);
        while(!q.empty()){
            int len=q.size();
            vector<int> rec;
            for(int i=0; i<len; i++){
                Node* node=q.front();
                q.pop();
                rec.push_back(node->val);
                vector<Node*> child=node->children;
                for(auto nc:child){
                    q.push(nc);
                }
            }
            ans.push_back(rec);
        }
        return ans;
    }
};

File: problems_list/0429.N叉树的层序遍历/0429-N叉树的层序遍历.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if(!root) return [];
    let ans=[];
    let queue=[root];
    while(queue.length){
        let n=queue.length;
        let layer=[];
        for(let i=0; i<n; i++){
            let node=queue.shift();
            layer.push(node.val);
            for(let ch of node.children){
                queue.push(ch);
            }
        }
        ans.push(layer);
    }
    return ans;
};